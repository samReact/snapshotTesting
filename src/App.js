import React, { Component } from "react";
import "./App.css";
import ShowOtherAdviceButton from "./\u001BShowOtherAdviceButton\u001B";

class App extends Component {
  state = {
    advice: ""
  };

  loadAdvice = async () => {
    const response = await fetch("https://api.adviceslip.com/advice");
    const {
      slip: { advice }
    } = await response.json();
    this.setState({ advice });
  };

  async componentDidMount() {
    await this.loadAdvice();
  }

  render() {
    return (
      <div className="App">
        <p className="Advice">{this.state.advice || "â€¦"}</p>
        <ShowOtherAdviceButton onClick={this.loadAdvice} />
      </div>
    );
  }
}

export default App;
