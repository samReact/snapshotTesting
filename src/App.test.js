import React from "react";
import App from "./App";
import { shallow, mount } from "enzyme";
import fetchMock from "fetch-mock";

describe("App", () => {
  beforeEach(() => {
    fetchMock.getOnce("https://api.adviceslip.com/advice", {
      slip: { advice: "Always trust random advice." }
    });
  });

  afterEach(() => {
    fetchMock.reset();
  });

  describe("before advice fetched", () => {
    it("renders loading indicator", () => {
      const appWrapper = shallow(<App />);
      expect(appWrapper).toMatchSnapshot();
    });
  });

  describe("after advice fetched", () => {
    it("renders advice", done => {
      const appWrapper = shallow(<App />);
      setImmediate(() => {
        expect(appWrapper).toMatchSnapshot();
        done();
      });
    });
  });

  describe('when hitting "Show other advice" button', () => {
    beforeEach(() => {
      fetchMock.getOnce(
        "https://api.adviceslip.com/advice",
        { slip: { advice: "Other random advice." } },
        { overwriteRoutes: false }
      );
    });

    it("should render other advice", done => {
      const appWrapper = mount(<App />);
      appWrapper.find("button").simulate("click");
      setImmediate(() => {
        appWrapper.update();
        expect(appWrapper).toMatchSnapshot();
        done();
      });
    });
  });
});
