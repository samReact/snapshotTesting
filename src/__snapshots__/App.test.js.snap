// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App after advice fetched renders advice 1`] = `
<div
  className="App"
>
  <p
    className="Advice"
  >
    Always trust random advice.
  </p>
  <ShowOtherAdviceButton
    onClick={[Function]}
  />
</div>
`;

exports[`App before advice fetched renders loading indicator 1`] = `
<div
  className="App"
>
  <p
    className="Advice"
  >
    â€¦
  </p>
  <ShowOtherAdviceButton
    onClick={[Function]}
  />
</div>
`;

exports[`App when hitting "Show other advice" button should render other advice 1`] = `
<App>
  <div
    className="App"
  >
    <p
      className="Advice"
    >
      Other random advice.
    </p>
    <ShowOtherAdviceButton
      onClick={[Function]}
    >
      <button
        className="Advice-other-advice-button"
        onClick={[Function]}
      >
        Show other advice
      </button>
    </ShowOtherAdviceButton>
  </div>
</App>
`;
